CMakePresets.json

```diff
@@ -29,24 +21,64 @@
     {
       "name": "gcc-debug",
       "displayName": "GCC Debug Build",
+      "inherits": ["_root-config"],
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "g++",
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_CXX_FLAGS": "-fsanitize=address -fsanitize=pointer-compare -fsanitize=pointer-subtract -fsanitize=leak -fsanitize=undefined"
+      }
+    },
+    {
+      "name": "gcc-release",
+      "displayName": "GCC Release Build",
       "inherits": [
         "_root-config",
-        "_debug-base"
+        "_release-base"
       ],
       "cacheVariables": {
         "CMAKE_CXX_COMPILER": "g++"
       }
     },
     {
-      "name": "gcc-release",
-      "displayName": "GCC Release Build",
+      "name": "xcode-debug",
+      "displayName": "xcode Debug Build",
+      "inherits": ["_root-config"],
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "g++",
```

Comment:
I think `g++` is the default compiler name for apple clang?

Reply:
I don't know what you mean by "default compiler name". AppleClang is installed on MacOS systems both to `/usr/bin/clang++` and `/usr/bin/g++`. `/usr/bin/clang++` is "more proper" than `/usr/bin/g++` because the latter is only there for compatibility with antiquated build systems that weren't aware of clang yet.

Reply:
It's a compatibility hack because too many build systems think there's always a /usr/bin/gcc when they're looking for the system compiler in /usr/bin/cc.

Reply:
Alright noted

Reply:
Adopted